test/parallel/test-runner-watch-coverage.mjs

```diff
@@ -0,0 +1,84 @@
+// Flags: --expose-internals
+import * as common from '../common/index.mjs';
+import { describe, it } from 'node:test';
+import { spawn } from 'node:child_process';
+import { writeFileSync } from 'node:fs';
+import util from 'internal/util';
+import tmpdir from '../common/tmpdir.js';
+import assert from 'node:assert';
+
+if (common.isIBMi)
+  common.skip('IBMi does not support `fs.watch()`');
+
+function getCoverageFixtureReport() {
+  const report = [
+    '# start of coverage report',
+    '# ---------------------------------------------------------------',
+    '# file           | line % | branch % | funcs % | uncovered lines',
+    '# ---------------------------------------------------------------',
+    '# dependency.js  | 100.00 |   100.00 |  100.00 | ',
+    '# dependency.mjs | 100.00 |   100.00 |  100.00 | ',
+    '# test.js        | 100.00 |   100.00 |  100.00 | ',
+    '# ---------------------------------------------------------------',
+    '# all files      | 100.00 |   100.00 |  100.00 |',
+    '# ---------------------------------------------------------------',
+    '# end of coverage report',
+  ].join('\n');
+
+  if (common.isWindows) {
+    return report.replaceAll('/', '\\');
+  }
+
+  return report;
+}
+
+tmpdir.refresh();
+
+const fixtureContent = {
+  'dependency.js': 'module.exports = {};',
+  'dependency.mjs': 'export const a = 1;',
+  'test.js': `
+const test = require('node:test');
+require('./dependency.js');
+import('./dependency.mjs');
+import('data:text/javascript,');
+test('test has ran');`,
+};
+
+const fixturePaths = Object.keys(fixtureContent)
+  .reduce((acc, file) => ({ ...acc, [file]: tmpdir.resolve(file) }), {});
+Object.entries(fixtureContent)
+  .forEach(([file, content]) => writeFileSync(fixturePaths[file], content));
+
+async function testWatch({ fileToUpdate, file }) {
+  const ran1 = util.createDeferredPromise();
+  const ran2 = util.createDeferredPromise();
+  const args = [ '--test', '--watch', '--experimental-test-coverage', file ? fixturePaths[file] : undefined];
+  const child = spawn(process.execPath,
+                      args.filter(Boolean),
+                      { encoding: 'utf8', stdio: 'pipe', cwd: tmpdir.path });
+  let stdout = '';
+
+  child.stdout.on('data', (data) => {
+    stdout += data.toString();
+    const testRuns = stdout.match(/ - test has ran/g);
+    if (testRuns?.length >= 1) ran1.resolve();
+    if (testRuns?.length >= 2) ran2.resolve();
+  });
+
+  await ran1.promise;
+  const content = fixtureContent[fileToUpdate];
+  const path = fixturePaths[fileToUpdate];
+  const interval = setInterval(() => writeFileSync(path, content), common.platformTimeout(1000));
```

Comment:
Please don't write the file sync here as well

Reply:
Are u suggesting to use [`writeFile`](https://nodejs.org/docs/latest/api/fs.html#fspromiseswritefilefile-data-options) instead `writFileSync`?? and any reason for this?

Reply:
I thought to avoid blocking